### DISCUSSION TITLE
Welcome to FiltaQuilla Discussions!

### DISCUSSION THREAD

**Discussion by RealRaven2000** [RealRaven2000/FiltaQuilla#278] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by TonyGravagno** [comment]
Thanks for creating this forum, Axel !!

[About me](https://about.me/tony.gravagno). I used Thunderbird years ago, got turned off by the black-box platform and issues. It just wasn't stable and continued support wasn't in the roadmap. A couple years ago I tried again and was pleased with updates. I recently saw a couple [vids by Alessandro Castellani](https://www.youtube.com/@alecaddd/search?query=thunderbird), and recent [Community Office Hours vids](https://www.youtube.com/@thunderbirdproject). It's apparent that this platform on the way back and I have confidence in its bright future.

I've been running my own mail servers for years but have decided to relieve myself of that stress and use quality hosted email services (tough to find). On my servers running Postfix, Dovecot, Pidgeonhole, and ManageSieve, I've enjoyed complex filtering, even though it's using the decades-old Sieve RFCs (server-side filtering). On a new host that I'm testing they're running Exim, which has very limited support for Sieve, and the Exim filters for users are very limited as well. That led me to the concession that I probably need to rewrite all of my filters in the client rather than on the server ... and that led me to FiltaQuilla and quickFilters.

It's really weird and cool that as soon as I got into these addons, I started to see familiar faces, names, and content everywhere: I have been a long-time fan of Alessandro for his vids on WordPress (I run many WP sites and write a lot of code) and other topics - but I had no idea that he's been associated with Thunderbird! 😲 Then I see Axel in vids, completely randomly, and John B, and others, and I suddenly felt like I was in the right place.

Looking at the coding aspects of Tb from an outside perspective, I've been confused and mildly turned off by a small developer community, so many references to old code that doesn't apply to modern development, so much focus on migration from old code to new, a large collection of APIs rather than a single interface point, the concept of "Experimental". As with many projects, the APIs seem to be documented from a JSDoc base, but I haven't found a reference that adequately describes when or why to reach for interfaces like nsImsgDbHdr. So, there's some chaos here, but I recognize the long-term legacy of this product and its migration to a new world by a dedicated group of committed people - so I'm buying in.

When I saw that Axel had adopted addons (from R. Kent) and is maintaining them in addition to his own, I got the sense of the community here. I appreciate his challenges to serve a diverse community for fun and profit. I completely recognize the "free is about liberty, not cost" mantra that most people still don't get. And I find the FOSS code here to be a good example of transitional software that allows me to understand how Thunderbird addons do what they do as I continue to learn about this ecosystem, its nuances and challenges.

Email is a decades-old paradigm based on hundreds of specs (RFCs) that should have died decades ago. But we're still stuck with it purely for lack of global motivation for change. So we need to make the best of it. I think filtering tools in particular are painfully primitive. The quickFilters and FiltaQuilla addons make this less painful. That's why I'm here.

For this Discussions area, I hope others will share their own desires for filtering email, challenges with these addons, and suggestions for enhancements. I hope I can translate some of my angst, long-term insight, and limited development energy into helping with some of this, ultimately "scratching my own itches" to eliminate even more pain from this awful ongoing chore.

Well Axel, you did ask for introductions. 🤣 

Who is next!?!?

**Comment by RealRaven2000** [comment]
Ok, I go next. I have developed quickFilters from scratch after prototyping its initial feature in the already very feature heavy QuickFolders. I decided to split into as separate code because the later was already getting very feature heavy (avoiding to use the term _bloated_). QuickFOlders I had joined i the very early days of v0.8 back in 2008 - I just added some cosmetic changes and then developed drag & drop for re-ordering the tabs. It was exciting to join an open source project at that time, imagine all the code open for everyone to improve! The original inventor eventually dropped out and left me to create a full product from it, eventually including necessary monetization in order to fund the ever expanding workload.

I love that we have these forums to discuss problems and new features and curate any additional function for the benefit of the majority of users. This also means sometimes to reject or simplify aspects in order to keep the user base content and not overwhelm them with feature creep. But I am usually very positive for new ideas that can be beneficial (and understood by the user base) as long as I can implement and support such features. Some items may take a little longer, especially here on FiltaQuilla as changes are kind of high risk, and the Add-on isn't funded directly by its users.

My approach for creating new functions is requirements driven, so that means before immediately spec'ing out new functions and methods or designing a new UI I will first discuss what needs the individual use case tries to cover, and sometimes find simpler or alternative ways to implement, so it is always requirements first, then functional specifications and then coding - at least that is it in my mind. One can waste quite a lot of time discussing functional variations and then even write code that doesn't meet the actual requirements of a user at all, so I will try to guide this in a structured way if I feel a functional rabbit hole is opening up. It all depends on the complexity of the task.

FiltaQuilla was taken over from the original developer Kent James (who is also a pioneer of Add-on monetization and gave me some great help with my own) after he had to retire for health reasons. In places it has quite deep ideas but it is definitely held back by the restrictions in the core app (Thunderbird) which hasn't really grown in the area of filtering with the growing needs of its users. I am hoping  we can change this by submitting some patches to core (I have done this before) but only once the code base has been converted to Github as planned. the current build system uses mercurial and is just too much of a pain to work with for me.

As regards collaborations we have a weekly Thunderbird Add-on developer meeting every Thursday on Zoom with other developer friends and John Bieling, which is open to other interested contributors, we also meet on matrix chat on a regular basis where there is an Add-on Developer chat specifically for Thunderbird. I can share this more easily when I am back from my holidays... here is the hub for the Tb Add-on Dev community

https://developer.thunderbird.net/add-ons/community

John was very instrumental in the development and improvement of the [mail extension API](https://webextension-api.thunderbird.net/en/stable/), which is an augmentation of the Firefox extension API, and it aiming to simplify and make more stable any development of Add-ons for the mail sector (mail apps based on Mozilla. namely the code bases mozilla-central and comm-central which can also be investigated from [searchfox.org](https://searchfox.org) - beware rabbit holes ahead!). In order not to have to understand and follow this very changing code the extension APIs offer a structured (but a lot more restricted) way of coding Add-on functionality. Also it is permissions based which gives more responsibility back to the end user.

Thankfully the API wasn't made mandatory as the only but we still have the possibility of adding "experimental Add-ons" which have experimental APIs that can access core code. FiltaQUilla is probably still 95% "experimental" because the API currently doesn't expose filter extendability, we are hoping this may come in the future.
